### ISSUE TITLE
Flag --no-vcs-ignore should not be required when watching a specific file

### ISSUE THREAD

**Issue by danielo515** [watchexec/watchexec#674]
Please delete this template text before filing, but you _need_ to include the following:

- Watchexec's version
- The OS you're using: MacOs
- A log with `-vvv --log-file` (if it has sensitive info you can email it at felix@passcod.name — do that _after_ filing so you can reference the issue ID)
- A sample command that you've run that has the issue

Given this command:

```sh
watchexec --no-vcs-ignore -w  ../../packages/doh/lib/config/monday/environments/local.ts npx ampt run check-config
```

If you try the more intuitive thing first, the file is ignored and nothing happens:

```sh
watchexec -w  ../../packages/doh/lib/config/monday/environments/local.ts npx ampt run check-config
```

It feels verbose the requirement to type `--no-vcs-ignore` given that I'm watching one specific file. It confused me a long time until I decided to try it.

**Comment by solson** [comment]
I hit this issue and found it very confusing today. I had to resort to very verbose logging to figure out why watchexec was completely ignoring changes to the single file path I explicitly told it to watch.

**Comment by TheBlek** [comment]
@passcod I would like to take a shot at this. From what I've gathered there are a couple ways to solve this: 1. Modify GlobsetFilterer to take in a whitelist of globs which to check before everything else; 2. Modify WatchexecFilterer to have that whitelist. I personally lean towards GlobsetFilterer modification (as it handles path filtering) but that would mean API breakage as I understand it (GlobsetFilterer::new would need to take in additional parameter). What do you think?

**Comment by passcod** [comment]
I'm fine with API breaks, go for it.

**Comment by passcod** [comment]
Fix released in 2.2.0
